hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/BaseSparkInternalRowReaderContext.java

```diff
@@ -123,4 +125,11 @@ private Object getFieldValueFromInternalRow(InternalRow row, Schema recordSchema
       return null;
     }
   }
+
+  @Override
+  public UnaryOperator<InternalRow> projectRecord(Schema from, Schema to) {
+    UnsafeProjection projection = HoodieInternalRowUtils.generateUnsafeProjectionAlias(AvroConversionUtils.convertAvroSchemaToStructType(from),
```

Comment:
I was referring to AvroConversionUtils.convertAvroSchemaToStructType which can be replaced by HoodieInternalRowUtils.getCachedSchema(schema) to avoid repeated parsing.